src/lib.ts

```diff
@@ -0,0 +1,744 @@
+type Avatar = {
+  emoji: string;
+  background: Color;
+};
+
+interface Color {
+  r: number;
+  g: number;
+  b: number;
+}
+
+export enum Usage {
+  Any = "any",
+  Identity = "identity",
+}
+
+const FNV_OFFSET_BASIS = BigInt("14695981039346656037");
+const FNV_PRIME = BigInt("1099511628211");
+
+const fnv1aHash = (input: string) => {
+  let hash = FNV_OFFSET_BASIS;
+
+  for (let i = 0; i < input.length; i++) {
+    hash ^= BigInt(input.charCodeAt(i));
```

Comment:
This is different from the corresponding Rust code. In Rust we iterate over the bytes. Here we iterate over the Unicode code points. This is different for strings containing unicode characters encoded in multiple bytes (e.g. `"¢"`). To avoid this we should turn `input` into a byte array first
```typescript
const bytes = Buffer.from(input, "utf8")
```
and then iterate over the bytes.

This issue doesn’t manifest itself currently because the inputs are all ASCII. But we should make an effort to make the implementations map for all allowed inputs.

Reply:
Good catch! I just assumed we'd never use it for anything non-ascii.

aaff30fd0702db4998cb4a69f49cc6a2278a738f